src/ui/src/components/forms/DatasourceForm/index.tsx

```diff
@@ -38,49 +40,87 @@ const DatasourceForm = () => {
   const [schema, setSchema] = useState<string>('');
   const [host, setHost] = useState<string>('');
   const [port, setPort] = useState<string>('');
+  const [project, setProject] = useState<string>('');
+  const [dataset, setDataset] = useState<string>('');
+  const [file, setFile] = useState<File | null>(null);
 
   const submitForm = async () => {
-    let body;
+
+    let config = {}
+
     if (datasourceType === DataSourceTypes.SNOWFLAKE) {
-      body = {
-        name: datasourceName,
-        type: datasourceType,
-        config: {
-          driver: datasourceType,
-          account: account,
-          user: user,
-          password: password,
-          warehouse: warehouse,
-          database: database,
-          schema: schema,
-        },
+      config = {
+        driver: datasourceType,
+        account: account,
+        user: user,
+        password: password,
+        warehouse: warehouse,
+        database: database,
+        schema: schema,
       };
     } else if (
       datasourceType === DataSourceTypes.POSTGRESQL ||
       datasourceType === DataSourceTypes.REDSHIFT
     ) {
-      body = {
-        name: datasourceName,
-        type: datasourceType,
-        config: {
-          user: user,
-          password: password,
-          host: host,
-          port: parseInt(port),
-          database: database,
-          schema: schema
-        },
+      config = {
+        user: user,
+        password: password,
+        host: host,
+        port: parseInt(port),
+        database: database,
+        schema: schema
+      };
+    } else if (datasourceType === DataSourceTypes.BIGQUERY) {
+      // TODO: Implement better validation
+      if(!datasourceName || !project || !dataset || !file) return;
```

Comment:
why add validation here but not on any others?

Reply:
because adding validation to other seemed out of scope of the PR, hence added it only to BigQuery.

Reply:
Yeah... having it or not having it with this implementation will still produce confusing behaviors either way so I suppose it really doesn't matter given that it's still not done in a way that handles user input correctly